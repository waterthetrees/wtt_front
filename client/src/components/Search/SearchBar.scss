.search-container {
  .search-input {
    padding: 6px;
    border: 1px solid #ddd;
    border-radius: 4px;
  }
}

.MuiAutocomplete-popper {
  .search-result {
    justify-content: space-between;
    padding: 12px;
    .left-container {
      display: flex;
      align-items: center;
    }
    .result-info {
      padding-left: 8px;
      line-height: 1.2;
    }
    .address {
      font-size: 0.83rem;
    }
  }
  .group-header {
    padding: 1rem;
  }
  .no-results {
    padding: 4px 12px;
    color: #888;
  }
}
