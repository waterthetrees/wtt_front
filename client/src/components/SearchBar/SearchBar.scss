.searchbar {
    display: flex;
    align-items: center;
    border: 1px solid #00000050;
    width: 100%;
    min-width: 265px;
    margin-bottom: 10px;
    background-color: white;
    
    &__input{
        background-color: transparent;
        border: none;
        width: 100%;
        height: 32px;
    }
    
    &__input:focus{
        outline: none;
    }

    &__input::-webkit-input-placeholder{
        color: '#00000050',
    }
}