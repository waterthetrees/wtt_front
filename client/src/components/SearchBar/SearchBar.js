import React from 'react';
import { Search } from '@/components/Icons';
import './SearchBar.scss';

export const SearchBar = (props) => {
  return (
    <div className="searchbar" style={{ ...props?.style?.div }}>
      <label htmlFor="searchbar">
        <Search
          sx={{
            fontSize: '24px',
            marginLeft: '10px',
            marginRight: '10px',
            fontFamily: 'Montserrat',
          }}
        />
      </label>

      <input
        id="searchbar"
        style={{ ...props.style.input }}
        value={props.search}
        onChange={props.onChange}
        placeholder={props.placeholder}
        className="searchbar__input"
      />
    </div>
  );
};

export const searchArray = (array, term) => {
  return array.filter((obj) => {
    for (const key in obj) {
      if (
        obj[key] !== null &&
        obj[key].toString().toLowerCase().includes(term.toLowerCase())
      ) {
        return true;
      }
    }
    return false;
  });
};
