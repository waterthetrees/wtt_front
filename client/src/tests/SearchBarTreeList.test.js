import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { SearchBar, searchArray } from '@/components/SearchBar/SearchBar';

/**
 * These snapshots are designed to catch _any_ change to UI components, so they
 * are by definition very brittle. If the changes are related to your changes
 * and look expected, update these snapshots with `npm test -- -u`
 */

const mockProps = {
  style: {
    div: {
      width: 'max-content',
      borderRadius: '4px',
      fontSize: 'large',
    },
    input: { borderRadius: '4px', width: '100%' },
  },
  search: 'Apple',
  onChange: jest.fn(),
  placeholder: 'Search trees',
};

describe('<Searchbar /> snapshot specs', () => {
  it('renders correctly', () => {
    const searchBar = render(<SearchBar {...mockProps} />);
    expect(searchBar).toMatchSnapshot();
  });

  test('renders correctly', () => {
    const { getByPlaceholderText } = render(<SearchBar {...mockProps} />);
    expect(getByPlaceholderText('Search trees')).toBeInTheDocument();
  });

  test('handles onChange event', () => {
    const handleChange = jest.fn();
    const { getByPlaceholderText } = render(
      <SearchBar placeholder="Search here" onChange={handleChange} />,
    );
    const input = getByPlaceholderText('Search here');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  test('input value is updated correctly', () => {
    const { getByPlaceholderText } = render(<SearchBar {...mockProps} />);
    const input = getByPlaceholderText('Search trees');
    expect(input.value).toBe('Apple');
  });
});

describe('searchArray', () => {
  const data = [
    { commonName: 'Golden Apple', dbh: 5 },
    { commonName: 'Pear', dbh: 20 },
    { commonName: 'Red Maple', dbh: 7 },
  ];

  test('filters array based on search term', () => {
    const searchTerm = 'Golden Apple';
    const filteredData = searchArray(data, searchTerm);
    expect(filteredData).toHaveLength(1);
    expect(filteredData[0].commonName).toBe('Golden Apple');
  });

  test('returns an empty array if no match found', () => {
    const searchTerm = 'Black Walnut';
    const filteredData = searchArray(data, searchTerm);
    expect(filteredData).toHaveLength(0);
  });
});
