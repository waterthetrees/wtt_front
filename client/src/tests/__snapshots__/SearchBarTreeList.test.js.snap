// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> snapshot specs renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 24px;
  margin-left: 10px;
  margin-right: 10px;
  font-family: Montserrat;
}

<body>
    <div>
      <div
        class="searchbar"
        style="border-radius: 4px; font-size: large;"
      >
        <label
          for="searchbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </label>
        <input
          class="searchbar__input"
          id="searchbar"
          placeholder="Search trees"
          style="border-radius: 4px; width: 100%;"
          value="Apple"
        />
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 24px;
  margin-left: 10px;
  margin-right: 10px;
  font-family: Montserrat;
}

<div>
    <div
      class="searchbar"
      style="border-radius: 4px; font-size: large;"
    >
      <label
        for="searchbar"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </label>
      <input
        class="searchbar__input"
        id="searchbar"
        placeholder="Search trees"
        style="border-radius: 4px; width: 100%;"
        value="Apple"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
